Remove SLIST(sys/queue.h) from API

safl opened the pull request:
Now, two things:

* Removes the "SLIST" data structure from the public API
* Adds a missing header-inclusion in the Windows backend

See commit-messages for details.

Baekalfen commented:
Before merging, give me a ping, so I can remove the special `SLIST` handling in the Cython bindings -- only an improvement.

ankit-sam commented:
Changes look good, As I don't have windows setup so I have a question if fio build and work on windows now? Maybe we can Rishabh for this.

safl commented:
> 

It should, at least that is the intent :) @rs-sam can you verify that this is sufficient for the needed compatibility? Our current CI does not verify this, so some manual checking is needed. What should work now is building upstream fio, with the upstream xNVMe I/O engine enabled.

safl commented:
> Before merging, give me a ping, so I can remove the special SLIST handling in the Cython bindings -- only an improvement.

@Baekalfen any ideas on the Cython failure on Debian Bookworm? I saw the error both before and after the ``SLIST`` handling was removed on this PR. Since Bookworm is the latest/testing version of Debian, I guess it is actually unrelated to the changes on this PR, and probably something we will see as other PRs run.

karlowich reviewed (approved):
Looks good!

safl commented:
> Before merging, give me a ping, so I can remove the special `SLIST` handling in the Cython bindings -- only an improvement.

@Baekalfen ping :) I removed mentions of SLIST and squashed them into the commit ``api,queue,python: remove SLIST(sys/queue.h) from API``, anything else needed in that area?

Baekalfen reviewed (approved):
You already found the things I wanted to remove in the Cython headers, so that's good.

The error you got is because the memory allocation is based a bit on chance. My guess is that it happens once everything 30 tests. You can merge this PR, and I'll make some changes to make it more tolerant in another PR.

safl commented:
> You already found the things I wanted to remove in the Cython headers, so that's good.
> 
> The error you got is because the memory allocation is based a bit on chance. My guess is that it happens once every 30 tests. You can merge this PR, and I'll make some changes to make it more tolerant in another PR.

That would be great. Thanks.

safl commented:
> > That would be great. Thanks. It would be preferable to get that testcase stabilized such that this PR does not go in with failed CI mark on it.
> 
> It will most likely succeed this time without any changes to the code.

Unfortunately it did not, I provided a comment on https://github.com/OpenMPDK/xNVMe/pull/183

Baekalfen commented:
> > > That would be great. Thanks. It would be preferable to get that testcase stabilized such that this PR does not go in with failed CI mark on it.
> > 
> > 
> > It will most likely succeed this time without any changes to the code.
> 
> Unfortunately it did not, I provided a comment on #183

I think you're referring to #188. I have posted a more effective solution in #189.

safl commented:
> 

Yeah, it was #188 i meant. Looking forward to #189
